genesyscloud/telephony_providers_edges_phonebasesettings/resource_genesyscloud_telephony_providers_edges_phonebasesettings.go

```diff
@@ -0,0 +1,217 @@
+	edgesAPI := platformclientv2.NewTelephonyProvidersEdgeApiWithConfig(sdkConfig)
+
+	log.Printf("Creating phone base settings %s", name)
+	phoneBaseSettings, _, err := edgesAPI.PostTelephonyProvidersEdgesPhonebasesettings(phoneBase)
+	if err != nil {
+		return diag.Errorf("Failed to create phone base settings %s: %s", name, err)
+	}
+
+	d.SetId(*phoneBaseSettings.Id)
+
+	log.Printf("Created phone base settings %s", *phoneBaseSettings.Id)
+
+	return readPhoneBaseSettings(ctx, d, meta)
+}
+
+func updatePhoneBaseSettings(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	name := d.Get("name").(string)
+	description := d.Get("description").(string)
+	phoneMetaBase := gcloud.BuildSdkDomainEntityRef(d, "phone_meta_base_id")
+	properties := gcloud.BuildBaseSettingsProperties(d)
+	id := d.Id()
+
+	phoneBase := platformclientv2.Phonebase{
+		Id:            &id,
+		Name:          &name,
+		PhoneMetaBase: phoneMetaBase,
+		Properties:    properties,
+		Lines: &[]platformclientv2.Linebase{
+			{
+				Name:         &name,
+				LineMetaBase: phoneMetaBase,
+			},
+		},
+		Capabilities: buildSdkCapabilities(d),
+	}
+
+	if description != "" {
+		phoneBase.Description = &description
+	}
+
+	sdkConfig := meta.(*gcloud.ProviderMeta).ClientConfig
+	edgesAPI := platformclientv2.NewTelephonyProvidersEdgeApiWithConfig(sdkConfig)
+
+	phoneBaseSettings, resp, getErr := edgesAPI.GetTelephonyProvidersEdgesPhonebasesetting(d.Id())
+	if getErr != nil {
+		if gcloud.IsStatus404(resp) {
+			return nil
+		}
+		return diag.Errorf("Failed to read phone base settings %s: %s", d.Id(), getErr)
+	}
+	(*phoneBase.Lines)[0].Id = (*phoneBaseSettings.Lines)[0].Id
+
+	log.Printf("Updating phone base settings %s", name)
+	phoneBaseSettings, resp, err := edgesAPI.PutTelephonyProvidersEdgesPhonebasesetting(d.Id(), phoneBase)
+	if err != nil {
+		return diag.Errorf("Failed to update phone base settings %s: %v", name, err)
+	}
+
+	log.Printf("Updated phone base settings %s", d.Id())
+
+	return readPhoneBaseSettings(ctx, d, meta)
+}
+
+func readPhoneBaseSettings(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*gcloud.ProviderMeta).ClientConfig
+	edgesAPI := platformclientv2.NewTelephonyProvidersEdgeApiWithConfig(sdkConfig)
+
+	log.Printf("Reading phone base settings %s", d.Id())
+	return gcloud.WithRetriesForRead(ctx, d, func() *retry.RetryError {
+		phoneBaseSettings, resp, getErr := edgesAPI.GetTelephonyProvidersEdgesPhonebasesetting(d.Id())
+		if getErr != nil {
+			if gcloud.IsStatus404(resp) {
+				return retry.RetryableError(fmt.Errorf("Failed to read phone base settings %s: %s", d.Id(), getErr))
+			}
+			return retry.NonRetryableError(fmt.Errorf("Failed to read phone base settings %s: %s", d.Id(), getErr))
+		}
+
+		cc := consistency_checker.NewConsistencyCheck(ctx, d, meta, ResourcePhoneBaseSettings())
+		d.Set("name", *phoneBaseSettings.Name)
+		if phoneBaseSettings.Description != nil {
```

Comment:
You can replace these if statements with the functions from the `resourcedata` package